__init__.py

```diff
@@ -201,27 +184,28 @@ def get_alarm_local(self, alarm=None, timestamp=None):
         return datetime.fromtimestamp(ts, default_timezone())
 
     def set_alarm(self, when, name=None, repeat=None):
+        time = when.replace(second=0, microsecond=0)
         if repeat:
-            alarm = self._create_recurring_alarm(when, repeat)
+            alarm = self._create_recurring_alarm(time, repeat)
             alarm = {
                 "timestamp": alarm["timestamp"],
                 "repeat_rule": alarm["repeat_rule"],
                 "name": name or ""
             }
         else:
             alarm = {
-                "timestamp": to_utc(when).timestamp(),
+                "timestamp": to_utc(time).timestamp(),
                 "repeat_rule": "",
                 "name": name or ""
             }
 
         for existing in self.settings["alarm"]:
             if alarm == existing:
                 self.speak_dialog("alarm.already.exists")
-                return
-
-        self.settings["alarm"].append(alarm)
-        self._schedule()
+                return None
+        else:
```

Comment:
Should this be indented one more level?

Reply:
No the else is connected to the `for`...but it is sort of redundant since it would return from within the if-statement...